### PR TITLE
[Mellanox] Add device files for SN5600

### PR DESCRIPTION
**Pull request by Junchao-Mellanox**
#### Why I did it

Add device files for new platform SN5600

#### How I did it

Add device files for new platform SN5600

#### How to verify it

Manual test

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205

### CODE REVIEW

**Review by keboliu — APPROVED** [review]

File: device/mellanox/x86_64-nvidia_sn5600-r0/platform_components.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{
+    "chassis": {
+        "MSN3800": {
```
Comment (liat-grozovik) [inline device/mellanox/x86_64-nvidia_sn5600-r0/platform_components.json:3]: MS3800?

Comment (Junchao-Mellanox) [reply device/mellanox/x86_64-nvidia_sn5600-r0/platform_components.json]: Fixed

**Review by liat-grozovik — APPROVED** [review]

**Pull request merged**